aten/src/ATen/native/cpu/CatKernel.cpp

```diff
@@ -7,6 +7,14 @@
 #include <ATen/cpu/vec/vec.h>
 #include <c10/util/irange.h>
 
+// suppress warnings triggered by #pragma unroll:
+// "warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]"
+#if defined(__clang__)
+#  pragma clang diagnostic push
+#  pragma clang diagnostic ignored "-Werror"
+#  pragma clang diagnostic ignored "-Wpass-failed"
+#endif
```

Comment:
Use convenience macros and never ignore `-Werror` 
```suggestion
C10_CLANG_DIAGNOSTIC_PUSH()
C10_CLANG_DIAGNOSTIC_IGNORE("-Wpass-failed")
```